azure-pipelines/jobs/ci-linux.yml

```
@@ -56,16 +57,29 @@ jobs:
             in(variables['Build.Reason'], 'PullRequest')
         )"
     continueOnError: true
+  # Below step's sourced from : https://github.com/dotnet/runtime/blob/4e72737a8ec246297bce916ad747e4a2fb1265ac/eng/pipelines/installer/jobs/base-job.yml#L455-L468
+  # Files may be owned by root because builds don't set user ID. Later build steps run 'find' in
+  # the source tree, which fails due to permissions in the 'NetCore*-Int-Pool' queues. This step
+  # prevents the failure by using chown to clean up our source tree.
+  - script: |
+      set -x
+      docker run --rm \
+        -v "$(Agent.BuildDirectory):/root/build" \
+        -w /root/build \
+        bash -c "chown -R $(id -u):$(id -g) *"
+    displayName: Update file ownership from root to build agent account
+    continueOnError: true
+    condition: succeededOrFailed()
 
   # Create reference packages tarball and upload to azure storage
   - script: |
       set -x
       df -h
-      $(docker.run) $(docker.src.map) $(docker.src.work) $(imageName) ./publish.sh $(BUILD.BUILDNUMBER) --ci \
+      pushd $(rootDirectory)/source-build-reference-packages/src
```

I know from reviewing this before that this `pushd` is ok without a `popd`, but a future reader may see this and think it's a mistake that `popd` is missing. To avoid the false bug flag, I'd switch this to a plain old `cd`.

```suggestion
      cd "$(rootDirectory)/source-build-reference-packages/src"
```

(And add quotes to handle spaces, in theory.)